src/bp/ui-studio/src/web/views/OneFlow/diagram/nodes/TriggerNode/index.tsx

```diff
@@ -0,0 +1,210 @@
+          {node.conditions?.map((condition, index) => (
+            <Fragment key={index}>
+              <NodeContentItem
+                className={cx(style.hasJoinLabel, {
+                  [style.active]: selectedCondition?.node?.id === node.id && index === selectedCondition?.index
+                })}
+                onEdit={() => editNodeItem?.(node, index)}
+              >
+                <span className={style.content}>{lang.tr(conditionLabels[condition.id], { ...condition })}</span>
+              </NodeContentItem>
+              <span className={style.joinLabel}>{lang.tr('and')}</span>
+            </Fragment>
+          ))}
+        </div>
+      )}
+    </NodeWrapper>
+  )
+}
+
+export class TriggerNodeModel extends BaseNodeModel {
+  public conditions = []
+  public activeWorkflow: boolean
+  public isNew: boolean
+
+  constructor({
+    id,
+    x,
+    y,
+    name,
+    onEnter = [],
+    next = [],
+    conditions = [],
+    activeWorkflow = false,
+    isNew = false,
+    isStartNode = false,
+    isHighlighted = false
+  }) {
+    super('trigger', id)
+    this.setData({ name, onEnter, next, isStartNode, isHighlighted, conditions, activeWorkflow, isNew })
+
+    this.x = this.oldX = x
+    this.y = this.oldY = y
+  }
+
+  setData({ conditions = [], activeWorkflow = false, isNew = false, ...data }) {
+    this.conditions = conditions
+    this.activeWorkflow = activeWorkflow
+    this.isNew = isNew
+
+    super.setData(data as any)
+  }
+}
+
+export class TriggerWidgetFactory extends AbstractNodeFactory {
+  private editNodeItem: (node: TriggerNodeModel, index: number) => void
+  private selectedNodeItem: () => { node: TriggerNodeModel; index: number }
+  private deleteSelectedElements: () => void
+  private getConditions: () => string
+  private getCurrentFlow: any
+  private updateFlowNode: any
+  private switchFlowNode: (id: string) => void
+  private addCondition: () => void
+
+  constructor(methods) {
+    super('trigger')
+
+    this.editNodeItem = methods.editNodeItem
+    this.selectedNodeItem = methods.selectedNodeItem
+    this.deleteSelectedElements = methods.deleteSelectedElements
+    this.getCurrentFlow = methods.getCurrentFlow
+    this.updateFlowNode = methods.updateFlowNode
+    this.getConditions = methods.getConditions
+    this.switchFlowNode = methods.switchFlowNode
+    this.addCondition = methods.addCondition
+  }
+
+  generateReactWidget(diagramEngine: DiagramEngine, node: TriggerNodeModel) {
+    return (
+      <TriggerWidget
+        node={node}
+        getCurrentFlow={this.getCurrentFlow}
+        editNodeItem={this.editNodeItem}
+        onDeleteSelectedElements={this.deleteSelectedElements}
+        updateFlowNode={this.updateFlowNode}
+        selectedNodeItem={this.selectedNodeItem}
+        getConditions={this.getConditions}
+        switchFlowNode={this.switchFlowNode}
+        addCondition={this.addCondition}
+      />
+    )
+  }
+
+  getNewInstance() {
+    // @ts-ignore
```

Comment:
Just curious why this ts-ignore

Reply:
Just copy/pasted from other components, I'm not even sure why we need that method, but it works 😂